_posts/2020-03-02-junior-dev-advice.markdown

```diff
@@ -0,0 +1,117 @@
+good salaries always come with expectations attached, and one of those
+will always be that you can work well in a team and communicate with
+others. If someone's going to pay you all of that money, they'll
+expect a lot of transparency in return. And that expectation will
+only grow as you acquire more and more experience. Your experience
+will have value, and your employers will expect you to share that
+value with those around you, *without* stepping on any toes in the
+process.
+
+Some more recommended reads:
+
+[How to Write a Git Commit Message](https://chris.beams.io/posts/git-commit/)
+[Technical Debt Isn't Just
+Technical](https://corgibytes.com/blog/2020/02/12/technical-debt-isnt-just-technical/)
+(communication solves a lot of problems up front!)
+[The Developer’s Edge: How To Become A Senior
+Developer](https://medium.com/zerotomastery/developers-edge-how-to-become-a-senior-developer-f1ec1738cf45)
+ (notice how few of these skills are technical!)
+
+# Stay humble
+
+It won't take long for a newly minted developer to learn that there
+are plenty of developers who have a bit of an ego problem. Aside from
+the obvious problems that this can cause with teamwork and
+professional behavior, losing perspective in this way can lead to
+skills starting to stagnate. When you take on a job as a developer,
+wether you realize it or not, you're implicitly taking on another
+responsibility - the responsibility of always being a student. The
+biggest difference between a newbie developer and a seasoned veteran
+is that the senior developer *has been a student for a lot longer.*
+But remaining a life long student requires you to understand the
+difference between confidence and ego, and commit to putting aside
+even the slightest shred of ego. Stay open minded, and willing to
+consider the possibility that your previous understanding of something
+was incomplete or mistaken. Recognize that no matter how much you've
+learned, it's only the base for the next thing that you're going to
+learn. And realize that no matter how far you've come, you can still
+make dumb mistakes - and program accordingly. In a field full of big
+personalities, stand out by staying humble.
+
+Some reminders to read over time:
+
+[The Greatest Developer Fallacy Or The Wisest Words You’ll Ever
+Hear?](https://skorks.com/2011/02/the-greatest-developer-fallacy-or-the-wisest-words-youll-ever-hear/)
+[The Ten Commandments of Egoless
+Programming](https://blog.codinghorror.com/the-ten-commandments-of-egoless-programming/)
+[Give it five minutes](https://signalvnoise.com/posts/3124-give-it-five-minutes)
+[Are you a 10x Programmer, or Just a
+Jerk?](https://thenewstack.io/10x-programmer-just-jerk/)
+[Beautiful Code](https://alarmingdevelopment.org/?p=79)
+[You're a Bad Programmer. Embrace
+It.](https://dzone.com/articles/youre-bad-programmer-embrace)
+[Teach Yourself Programming in Ten Years](http://www.norvig.com/21-days.html)
```

Comment:
@rdormer It would be nice to have some kind of conclusion or recap at the end of the article